resolver-dns/src/main/java/io/netty/resolver/dns/DnsServerAddresses.java

```diff
@@ -273,24 +259,21 @@ public void remove() {
         return list.toArray(new InetSocketAddress[list.size()]);
     }
 
-    private DnsServerAddresses() { }
+    protected DnsServerAddresses() { }
 
-    private static final class SequentialAddressIterator implements Iterator<InetSocketAddress> {
+    public abstract DnsServerAddressStream stream();
+
+    static final class SequentialDnsServerAddressStream implements DnsServerAddressStream {
```

Comment:
It is also being used by RotationalDnsServerAddresses. Let me move these classes to top level so it's less crowded here btw.